src/render.rs

```diff
@@ -0,0 +1,602 @@
+        );
+
+        let ts = ThemeSet::load_defaults();
+        let mut prev_syntax = ss.find_syntax_plain_text() as *const _;
+        let theme = ts.themes.get(theme).with_context(|| {
+            format!(
+                "Could not find theme {theme:?}, must be one of {}",
+                ts.themes
+                    .keys()
+                    .map(|s| format!("{s:?}"))
+                    .collect::<Vec<_>>()
+                    .join(", ")
+            )
+        })?;
+
+        let threads = (threads == 0).then(num_cpus::get).unwrap_or(threads);
+        let (mut line_num, longest_line_chars, background) = if threads < 2 {
+            let mut line_num: u32 = 0;
+            let mut longest_line_chars = 0;
+            let mut background = None;
+            let mut highlighter =
+                syntect::easy::HighlightLines::new(ss.find_syntax_plain_text(), theme);
+            for ((path, content), num_content_lines) in content {
+                progress.inc();
+                if should_interrupt.load(Ordering::Relaxed) {
+                    bail!("Cancelled by user")
+                }
+
+                if !plain {
+                    let syntax = ss
+                        .find_syntax_for_file(&path)?
+                        .unwrap_or_else(|| ss.find_syntax_plain_text());
+                    if syntax as *const _ != prev_syntax {
+                        highlighter = syntect::easy::HighlightLines::new(syntax, theme);
+                        prev_syntax = syntax as *const _;
+                    }
+                }
+
+                if display_to_be_processed_file {
+                    progress.info(format!("{path:?}"))
+                }
+                let out = chunk::process(
+                    &content,
+                    &mut img,
+                    |line| highlighter.highlight_line(line, &ss),
+                    chunk::Context {
+                        column_width,
+                        line_height,
+                        total_line_count,
+                        highlight_truncated_lines,
+                        line_num,
+                        lines_per_column,
+                        fg_color,
+                        bg_color,
+                    },
+                )?;
+                longest_line_chars = out.longest_line_in_chars.max(longest_line_chars);
+                line_num += num_content_lines as u32;
+                line_progress.inc_by(num_content_lines);
+                background = out.background;
+            }
+
+            (line_num, longest_line_chars, background)
+        } else {
+            let mut line_num: u32 = 0;
+            let mut longest_line_chars = 0;
+            let mut background = None;
+            std::thread::scope(|scope| -> anyhow::Result<()> {
+                let (tx, rx) = flume::bounded::<(_, String, _, _)>(content.len());
+                let (ttx, trx) = flume::unbounded();
+                for tid in 1..threads {
```

Comment:
Shouldn't this be `0..threads`?

Reply:
It does count the main thread as working thread as it aggregates, thus it excludes it. It's probably OK to overcommit a little but I wanted to be sure the aggregator will not be starved as it will slow down all producers.